R/manual_layer_udf.R

```diff
@@ -22,10 +22,15 @@
 ##' position, use a list like \code{args=list(123, 456)}. If you want
 ##' to call by name, use a named list like \code{args=list(x=123,y=456)}.
 ##'
+##' @param result_format One of \code{native}, \code{json}, or \code{arrow}. These are
+##' used as wire format for returning results from the server to this library, primarily
+##' for memory-usage control.  UDF return values handed back to your code from this
+##' library are converted back to natural R objects.
+##'
 ##' @return The R object which is the return value from the UDF.
 ##' @family {manual-layer functions}
 ##' @export
-execute_generic_udf <- function(namespace, udf=NULL, registered_udf_name=NULL, args=NULL) {
+execute_generic_udf <- function(namespace, udf=NULL, registered_udf_name=NULL, args=NULL, result_format='native') {
```

Comment:
Not stricly needed but I have found over the years that adding 

```r
result_format <- match.arg(result_format)
```

after you declare the signature as

```r
result_format=c("native", "json", "arrow"))
```

helps a _ton_.  The function will cleanly error for you if you submit, say `result_format="rara" and will do partial matching (can be useful) and will not get in the way otherwise.  The first value in the signature vector is the default, so still `native` here.

Reply:
@eddelbuettel nice!

I was (without saying so) relying on deeper-level throw in
https://github.com/TileDB-Inc/TileDB-Cloud-R/blob/master/R/result_format.R#L29-L31
via `ResultFormat$new(result_format)`

-- I like the idea of taking your suggestion here too, surfacing the error a little higher up; however, I'm reluctant to 'double-enter' the enum values as the ones in `R/result_format.R` are canonical & come from OpenAPI autogen.

I think for this PR I'd like to stick with the `ResultFormat$new` match-checker, & just better comment here that I'm relying on it to do just that ...